src/IssueViz.Security/ObservingIssueLocationStore.cs

```diff
@@ -0,0 +1,123 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security
+{
+    internal interface IObservingIssueLocationStore : IIssueLocationStore, IDisposable
+    {
+        void Register(ObservableCollection<IAnalysisIssueVisualization> issueVisualizations);
+        void Unregister(ObservableCollection<IAnalysisIssueVisualization> issueVisualizations);
+    }
+
+    [Export(typeof(IObservingIssueLocationStore))]
+    [Export(typeof(IIssueLocationStore))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class ObservingIssueLocationStore : IObservingIssueLocationStore
+    {
+        private IList<ObservableCollection<IAnalysisIssueVisualization>> ObservableIssueVisualizations { get; } = new List<ObservableCollection<IAnalysisIssueVisualization>>();
+
+        public void Register(ObservableCollection<IAnalysisIssueVisualization> issueVisualizations)
```

Comment:
Assuming `IObservingIssueLocationStore` doesn't need to implement `IIssueLocationStore`, could you make it clearer which methods belong to which interface? i.e. explicit implementation or regions.

That would make it easier to see the two distinct faces of this class - the "input" side used by the stores that provide the data, and the "output" side used by the consumers.